### DISCUSSION TITLE
Extract on output the "found language"?

### DISCUSSION THREAD

**Discussion by islogged** [openai/whisper#1601] [General]
Hey all,
I use whisper via the command line...

Is there a way with whisper to get in a file the language found?
When it detect "English" or "Spanish" it display it in the console, but i need to extract this information and write it in a file?

Is there a way?

Big thx to all...

**Comment by glangford** [comment]
If you use `--verbose True` from the command line, it will print out the detected language to the console. To get that saved to a file, redirect the output. For example,

`whisper test.mp3 --model small --verbose True >log.txt`

Otherwise, if you use Python you can get the detected language and save it; see the README for an example.

**Reply by islogged** [reply]
First thank you glangford for your answer.

> Otherwise, if you use Python you can get the detected language and save it; see the README for an example

I don't use, and i don't know to use Python.

> If you use --verbose True from the command line, it will print out the detected language to the console. To get that saved to a file, redirect the output. For example,
> 
> whisper test.mp3 --model small --verbose True >log.txt

**Yeah, that's what i finally did!**
But i lose the "live transcribe" view in the console. :(

I tried "powershell ".\something.exe | tee test.txt" but it show the result in the console only at the end of the process and make some problems with special characters ( é, è, ê, etc ) :(

If you know a way for this: https://stackoverflow.com/questions/796476/displaying-windows-command-prompt-output-and-redirecting-it-to-a-file I will take it!

Thx again...

**Reply by glangford** [reply]
@islogged I am not a Windows or a Powershell user, but you could try one of these ideas

* `something.exe | type test.txt` (suggested here https://stackoverflow.com/questions/503846/how-do-i-echo-and-send-console-output-to-a-file-in-a-bat-script)
* `tee-object` with a `-Encoding` argument for the special characters (https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/tee-object?view=powershell-7.3)
(but I think `tee-object` may be just an alias for `tee` so I don't know that it would fix the live transcribe)

**Reply by islogged** [reply]
Thank you for ur answer. I tried it all, but nothing work ...

There is no character problem if i use the powershell console, but for some reason too long to explain i need to launch the command in the cmd console.

Anyway tee or tee-object display the result when all the transcribe is finish, not "line by line" as whisper normally does ...

ChatGPT also gave me many examples to fix this, but nothing worked for me... 

Anyway all of these is not a problem of Whisper itself!

Thank anyway,
